Classes/Domain/Repository/HaikuRepository.php

```diff
@@ -32,8 +34,11 @@ public function findByUid(int $uid): array
         $where = $queryBuilder->expr()->eq('uid', $uid);
         $result = $queryBuilder->select('*')->from(self::TABLENAME)->where(
             $where
-        )->executeQuery();
-        return $result->fetchAssociative() ?? [];
+        )->executeQuery()->fetchAssociative();
+        if (!$result) {
```

Comment:
I suggest to use `$result !== []` as condition. This way it is clear, that another result as an array is an error. It's more obvious ;-) Same in line 56.

Reply:
No that wouldn't make sense the result would be false in case of a failure and a non empty array otherwise

Reply:
You are right, shouldn't make code reviews after a long day....